### PR TITLE
Rename shuffle to shuffle_method in remaining methods

### LINKED ISSUE #10782: Rename shuffle to shuffle_method in remaining methods
Mostly groupby aggregations and drop_duplicates (maybe unique and nunique)

please grep the code base.

### PR DESCRIPTION
**Pull request by milesgranger**
- [x] Closes #10782
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

---

Is there suppose to be a deprecation period before this renaming?

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -209,9 +209,9 @@ def _determine_split_out_shuffle(shuffle, split_out):
             return config.get("dataframe.shuffle.method", None) or "tasks"
         else:
             return False
-    if shuffle is True:
+    if shuffle_method is True:
```
Comment (phofl) [inline dask/dataframe/core.py:212]: That is something we want to deprecate as well, shuffle_method can only be a string the determines the shuffle method

split_out should be used to signal if you want to shuffle or not

Comment (phofl) [reply dask/dataframe/core.py:212]: We can do this as a follow up though

Comment (milesgranger) [reply dask/dataframe/core.py:212]: Ya, if it's good for you, would like to do this in a follow up.

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -4891,15 +4891,15 @@ def drop_duplicates(
         self,
         split_every=None,
         split_out=1,
-        shuffle=None,
+        shuffle_method=None,
```
Comment (phofl) [inline dask/dataframe/core.py:4894]: We have to deprecate here, there is a nice decorator that does this for you

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

**Pull request merged**